### DISCUSSION TITLE
Immovable Objects

### DISCUSSION THREAD

**Discussion by TKDKid1000** [FabricMC/fabric-api#1616] [Mod Dev Support]
Anyone know a simple way to make a block immovable? Trying to add a new redstone device.

**Comment by warjort** [comment]
Your new device is a BlockEntity? They can't be moved by pistons.

Otherwise;
````
new MyNewBlock(
  AbstractBlock.Settings.of(
    new FabricMaterialBuilder(MapColor.whatever).pistonBehavior(PistonBehavior.BLOCK).build()
  )
);
````

**Reply by TKDKid1000** [reply]
It is not a block entity, think immovable obsidian slab.

**Comment by haykam821** [comment]
Override the `getPistonBehavior` method in your block class:

```java
@Override
public PistonBehavior getPistonBehavior(BlockState state) {
	return PistonBehavior.BLOCK;
}
```
